test/web-platform-tests/to-run.yaml

```diff
@@ -800,8 +803,7 @@ leaktests/html-collection.html: [fail, Document.all is not implemented]
 leaktests/window-frames.html: [fail, Window.name is not implemeneted]
 scroll-to-the-fragment-in-shadow-tree.html: [fail, Requires a layout engine]
 slotchange-customelements.html: [fail, CustomElement.define is not implemented]
-slotchange-event.html: [timeout, MutationObserver is not implemented]
-slotchange.html: [timeout, MutationObserver is not implemented]
+slotchange-event.html: [timeout, Compound microtask scheduling issue]
```

Comment:
This is an interesting aspect of this PR. Is that since we are using a microtask to simulate the behavior of a compound microtask we run into some scheduling issues between `MutationObserver` and `slotchange` events.

I don't see how to get around this issue without having to fully reimplement an event-loop in the context of jsdom.

Reply:
Can you say a bit more? I was vaguely under the impression that we could implement compound microtasks by posting a single microtask that does several things.

Reply:
I was wrong for the microtask here. 

If you look at the test: `slotchange event must fire at the end of current microtask after mutation observers are invoked`:
1. Adds 2 mutation observers (one on `id` attribute and another one on `title`) and a slot with a `slotchange` event handler.
2. Update the `id` attribute and the append a child in the light dom of the host. We then enqueue a microtask to notify the listeners. At this point the `id` attribute mutation observer contains 1 mutation record, and the signal slot list contains the slot.
3. In the next microtask, the notifiy observer algorithm is executed:
	1. The mutation observer list is copied, it contains the 2 mutation observers, and it iterates over the 2 mutation observers
	2. The callback for the `id` mutation observer is executed and in it's callback it updates title attribute and append a new element the light dom of the host. By updating the `title` attribute a new mutation record is enqueued but this time for the `title` mutation observer.
	3. The callback fro the `title` mutation observer is invoked because it has a mutation record. And the test throws here since it expects that the `slotchange` event is invoked first.

I don't understand, how this test can pass according to the specification. [Chrome](https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/dom/mutation_observer.cc?sq=package:chromium&g=0&l=349-350) and [Webkit](https://trac.webkit.org/browser/webkit/trunk/Source/WebCore/dom/MutationObserver.cpp#L273) don't run into the same issue since they keep track of an active mutation observer list instead of iterating over all the mutation observers.

The test pass if we introduce a similar logic in `notifyMutationObservers` by only iterating over the mutations observers with mutation records.

```js
const notifyList = mutationObserverList.filter(mo => {
    return mo._recordQueue.length;
});
```

Reply:
Very interesting! Sounds like it's worth opening a spec issue? And probably introducing our own similar logic for now, so that we behave similar to browsers?